p2p/p2p.go

```diff
@@ -346,3 +367,109 @@ func (s *Service) WithListener(l junoSync.EventListener) {
 	runMetrics(s.host.Peerstore())
 	s.synchroniser.WithListener(l)
 }
+
+// EncodeAddrs encodes a slice of multiaddrs into a byte slice
+func EncodeAddrs(addrs []multiaddr.Multiaddr) ([]byte, error) {
+	multiAddrBytes := make([][]byte, len(addrs))
+	for i, addr := range addrs {
+		multiAddrBytes[i] = addr.Bytes()
+	}
+
+	var buf bytes.Buffer
+	if err := cbor.NewEncoder(&buf).Encode(multiAddrBytes); err != nil {
+		return nil, fmt.Errorf("encode addresses: %w", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+// DecodeAddrs decodes a byte slice into a slice of multiaddrs
+func DecodeAddrs(b []byte) ([]multiaddr.Multiaddr, error) {
```

Comment:
can we please make both of these functions unexported and move everything that relates to peers into `peers.go` file (p2p package)

Reply:
In my test case, I need to use `EncodeAddrs` it needs to be exportable.